core/src/main/java/org/apache/calcite/jdbc/CalciteMetaImpl.java

```diff
@@ -90,13 +90,40 @@
 public class CalciteMetaImpl extends MetaImpl {
   static final Driver DRIVER = new Driver();
 
+  private final CalciteMetaTableFactory metaTableFactory;
+  private final CalciteMetaColumnFactory metaColumnFactory;
+
+  /** Creates a CalciteMetaImpl.
+   *
+   * @deprecated Use
+   * {@link #CalciteMetaImpl(CalciteConnectionImpl, CalciteMetaTableFactory, CalciteMetaColumnFactory)}
+   * instead.
+   */
+  @Deprecated // to be removed before 2.0
   public CalciteMetaImpl(CalciteConnectionImpl connection) {
+    this(connection, new CalciteMetaTableFactoryImpl(), new CalciteMetaColumnFactoryImpl());
+  }
+
+  public CalciteMetaImpl(CalciteConnectionImpl connection,
```

Comment:
Add javadoc.

I would remove the `@Nullable`. Caller can easily provide factories.

Reply:
It's the Java 11 CI that is giving me issues when I remove the @nullable annotation. 

https://github.com/apache/calcite/actions/runs/6255263301/job/16984285491?pr=3421

Reply:
I think CI is finding cases where you might accidentally supply null (e.g. when `Driver` calls `config.metaColumnFactory`). I'll fix.